packages/t2l-backend/src/externalApis/ladokApi/index.ts

```
@@ -85,10 +85,22 @@ export function getSkaFinnasStudenter(aktivitetstillfalleUID: string) {
  * Get the structure of a course round (kurstillfälle)
  * @see {@link https://www.integrationstest.ladok.se/restdoc/resultat.html#h%C3%A4mtaIng%C3%A5endeMomentF%C3%B6rKurstillf%C3%A4lle}
  */
-export function getKurstillfalleStructure(kurstillfalleUID: string) {
+export async function getKurstillfalleStructure(kurstillfalleUID: string) {
   return gotClient
     .get<Kurstillfalle>(`resultat/kurstillfalle/${kurstillfalleUID}/moment`)
-    .then((response) => response.body);
+    .then((response) => response.body)
+    .catch((e) => {
+      const errorBody = e.response.body;
+      if (
+        errorBody.Detaljkod === "resultat.fel.detaljkod.utbildningstillfalle" &&
+        errorBody.Felgrupp === "commons.fel.grupp.obligatoriskt_varde_saknas"
+      ) {
+        // This looks like the course round doesn't exist in Ladok. Ignore it.
+        return null;
```

We should probably log this, it's a bit dangerous to silently skip things

---

I am not sure if the function should return `null` in this case or throw a more detailed error, something like `LadokApiError` or something like that, but that can be done in the future.

The function spec (lines 85-86) is updated.

Agree to log things, but definitely _not_ in this module. As described in line 2:

```
* This module are functions to call Ladok API. They do not contain any logic.
```